server/sandbox_run_linux.go

```diff
@@ -320,20 +320,28 @@ func (s *Server) getSandboxIDMappings(ctx context.Context, sb *libsandbox.Sandbo
 		return nil, errors.New("infra container not found")
 	}
 
-	uids, err := rootless.ReadMappingsProc(fmt.Sprintf("/proc/%d/uid_map", ic.State().Pid))
+	uids, gids, err := unshare.GetHostIDMappings(strconv.Itoa(ic.State().Pid))
 	if err != nil {
 		return nil, err
 	}
-	gids, err := rootless.ReadMappingsProc(fmt.Sprintf("/proc/%d/gid_map", ic.State().Pid))
-	if err != nil {
-		return nil, err
-	}
-
-	mappings := idtools.NewIDMappingsFromMaps(uids, gids)
+	mappings := convertToMappings(uids, gids)
 	ic.SetIDMappings(mappings)
 	return mappings, nil
 }
 
+func convertToMappings(uids, guids []spec.LinuxIDMapping) *idtools.IDMappings {
```

Comment:
Can you make this function look more like the one above? The `convertToStorageIDMap()`. Especially as you know the number of UIDs and GIDs, etc.  Also, you could move, so both functions are closer together.

Reply:
Which one are you suggesting?

I have to do this conversion because the rootless and unshare code have slightly different types.

Reply:
This function is one line past the end. Not sure how it can get closer. I could define it in the function if you'd prefer

Reply:
This is what I had in mind (an example):

```go
func convertToStorageIDMappings(uidMappings, gidMappings []spec.LinuxIDMapping) *idtools.IDMappings {
	uids := make([]idtools.IDMap, len(uidMappings))
	gids := make([]idtools.IDMap, len(gidMappings))

	for i, v := range uidMappings {
		uids[i] = idtools.IDMap{
			ContainerID: int(v.ContainerID),
			HostID:      int(v.HostID),
			Size:        int(v.Size),
		}
	}
	for i, v := range gidMappings {
		gids[i] = idtools.IDMap{
			ContainerID: int(v.ContainerID),
			HostID:      int(v.HostID),
			Size:        int(v.Size),
		}
	}

	return idtools.NewIDMappingsFromMaps(uids, gids)
}
```

To make it look more like the other.  So, avoid `append()`s there, since we know the length in advance, etc.

Reply:
Holding to address your comment. Will do that today.

Reply:
Updated.